### PR TITLE
Re-enable REMD with TC-MPI interface

### PR DESCRIPTION
**Pull request by danielhollas**
This has worked before but was removed in the previous commit. Here I add it back and add a test.

The current approach is a bit hacky, and requires that the number of TC servers is exactly equal to number of replicas ,
as each MPI process is connecting to its own TC server.
This is untenable for larger number of replicas so we might need to improve upon this in the future.
(note that for technical reasons, `nteraservers` must be 1 in the ABIN input). Though maybe we should just ignore the setting and set it automatically in the code?)

### CODE REVIEW

File: src/init.F90 [hunk 1]
```diff
@@ -622,22 +616,33 @@ subroutine init(dt)
...
       write (*, *)
-      if (inose >= 1) write (*, nml=nhcopt)
-      write (*, *)
-      if (ipimd == 2 .or. ipimd == 4) write (*, nml=sh)
-      write (*, *)
-      if (ipimd == 5) write (*, nml=lz)
-      write (*, *)
-      if (iqmmm == 3 .or. pot == 'mm') write (*, nml=qmmm)
-      write (*, *)
+      if (inose >= 1) then
```
Comment (danielhollas) [inline src/init.F90:619]: Unrelated cleanup. In general I am getting rid of one line if statements since they don't play nicely with code coverage/

File: src/init.F90 [hunk 2]
```diff
@@ -622,22 +616,33 @@ subroutine init(dt)
...
+      end if
+      if (iqmmm == 3 .or. pot == 'mm') then
+         write (*, nml=qmmm)
+         write (*, *)
+      end if
    end if
 #ifdef USE_MPI
    call MPI_Barrier(MPI_COMM_WORLD, ierr)
    write (*, '(A,I0,A,I0)') 'MPI rank: ', my_rank, ' PID: ', GetPID()
+   call MPI_Barrier(MPI_COMM_WORLD, ierr)
```
Comment (danielhollas) [inline src/init.F90:639]: Just for a nicer output.

File: tests/TERAPI-REMD/input.in [hunk 3]
```diff
@@ -0,0 +1,37 @@
+
```
Comment (danielhollas) [inline tests/TERAPI-REMD/input.in:1]: I took this file and the reference files from the `REMD` test.

**Comment by danielhollas** [comment]
@suchanj can you take a look? The only (small) functional change is in `tera_mpi_api.F90`. Then there are just a couple of cosmetic changes in other files and a whole bunch of new test files, but they were mostly copied from existing tests so nothing interesting to review there.

**Review by suchanj — APPROVED** [review]

**Comment by danielhollas** [comment]
@suchanj I am trying to test this on NEON but with no luck. :-(
I am using the `r.terabin` script, but when I submit it, it is stuck on the node, and the standard outputs from ABIN and TC say only:

```
error: executing task of job 332517 failed: execution daemon on host "n28.vscht.cz" didn't accept task
```
It looks like some bad interaction between MPI and SGE, but this is the first time that I see this. I've tried multiple MPICH versions with no luck. Have you encountered this before?

I was only able to test on `n28`, `n26` is throwing the `SGE CUDA starter` error, and the other nodes are full.

**Comment by suchanj** [comment]
@danielhollas I haven't seen this error before. It is also not very specific... Well, this functionality is not critical so we might wait for new cluster to be installed. I reported the n26 SGE CUDA error. (PS: I also remember having minor doubts about future r.terabin functionality based on previous commits, just don't remember why.)

**Comment by danielhollas** [comment]
@suchanj thanks. I'm going to merge this but will need to figure it out. The problem is that I am not able to run normal MD either. It might also be an issue with the fact the TeraChem is compiled with different MPICH version, I am not sure.
